docs/source/library/Library.rst

```diff
@@ -0,0 +1,16 @@
+Library
+=======
+
+.. toctree::
+    :maxdepth: 4
+
+    bitset
+    circuit
+    gate
+    gate_type
+    program
+    properties
+    read_program_settings
+    simple_simulation
+    cost_aware_synthesis
+    line_aware_synthesis
```

Comment:
This could use a lot more structure. See, e.g. the API documentation of QMAP or QCEC.
In this case, this could definitely be split into 
- something about SyReC itself and the corresponding parser
- something about simulation
- something about synthesis
- something about circuits (and gates)

Also, each of the files could probably use some more documentation. Either directly in the files other via docstrings in the bindings.cpp file.

Reply:
I have rearranged the order. I will add more documentation via docstrings in the following commit

Reply:
done 👍🏻